### PR TITLE
Macro language for defining structures

### PR DESCRIPTION
**Pull request by ecavallo**

### CODE REVIEW

**Comment by mortberg** [comment]
Wow, this is super cool!

File: Cubical/Structures/Macro.agda [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+open import Cubical.Foundations.SIP renaming (SNS-PathP to SNS)
+open import Cubical.Functions.FunExtEquiv
+open import Cubical.Data.Sigma
+
+open import Cubical.Structures.Constant
+open import Cubical.Structures.Pointed
+open import Cubical.Structures.NAryOp
+open import Cubical.Structures.Parameterized
+
+data Desc (ℓ : Level) : Typeω where
```
Comment (mortberg) [inline Cubical/Structures/Macro.agda:17]: It's annoying that this is in `Typeω`... But not much to do about it I guess?

Comment (mortberg) [reply Cubical/Structures/Macro.agda:17]: Some comments explaining the various field would be helpful. Especially the latter ones

Comment (ecavallo) [reply Cubical/Structures/Macro.agda:23]: I added some comments.

**Pull request merged**